src/positional.rs

```diff
@@ -36,41 +36,68 @@ pub struct Positional {
   pub arguments: Vec<String>,
 }
 
+#[derive(Copy, Clone)]
+enum ProcessingStep {
+  Overrides,
+  SearchDir,
+  Arguments,
+}
+
 impl Positional {
-  pub fn from_values<'values>(values: Option<impl IntoIterator<Item = &'values str>>) -> Self {
+  pub(crate) fn from_values<'values>(
+    values: Option<impl IntoIterator<Item = &'values str>>,
+  ) -> Result<Self, ConfigError> {
     let mut overrides = Vec::new();
     let mut search_directory = None;
     let mut arguments = Vec::new();
 
+    let mut processing_step = ProcessingStep::Overrides;
+
```

Comment:
This might be cleaner with something like:

```
let values = values.into_iter().peekable();

while let Some(value) = values.peek() {
  match processing_step {
    // do step, using values.next() to advance iterator
  }
}
```

Not sure though, TIOLI.

Reply:
I coded this up - I think it's more or less as clean (which is to say, slightly messy) as the previous solution. Might as well use it since it's added to this PR now unless you really hate it.